### DISCUSSION TITLE
Inbuild methods like copy (), clone(), compare () etc

### DISCUSSION THREAD

**Discussion by Munsif1** [pyuvm/pyuvm#136] [Q&A]
HI,

Sorry if I'm repeating this question, if already been asked.

Are there any inbuilt copy(), close (), compare (), etc methods, as we have in case of sv-uvm?

**Comment by raysalemi** [comment]
> HI,
> 
> 
> 
> Sorry if I'm repeating this question, if already been asked.
> 
> 
> 
> Are there any inbuilt copy(), close (), compare (), etc methods, as we have in case of sv-uvm?

Yes.

As with sv-uvm, you must override `do_copy()` and `do_compare()` and call `super()`

```

recording')

    # 5.3.8.1
    def copy(self, rhs):
        """
        Copy fields from rhs to this object
        """
        self.do_copy(rhs)

    # 5.3.8.2
    def do_copy(self, rhs):
        """
        Copies name. Override to copy additional data members
        """
        self.set_name(rhs.get_name())

    # 5.3.9.1
    def compare(self, rhs):
        """

        """
        return self.do_compare(rhs)

    # 5.3.9.2
    def do_compare(self, rhs):
        """
        Recommend overriding __eq__() rather than this method.
        """
        return self.__eq__(rhs)

 ```

**Reply by Munsif1** [reply]
Thanks for the reply:)

So we have to define our own do_copy() and do_compare(), do_print(), and make use of super().

can't we do something like as below, in pyuvm.

seq_itm_1.copy(seq_itm_0);					// copy seq_itm_0 into seq_itm_1 (object.copy)
seq_itm_1.print();							// and print seq-itm_1, (object.print()) 


// full code   https://edaplayground.com/x/eZfr

**Comment by raysalemi** [comment]
No. Print is not supported, but you are free to implement it.

I never use print() or teach it because it requires field macros in SV.

**Reply by Munsif1** [reply]
Ok, Thanks..
